hadoop/src/main/java/org/apache/carbondata/hadoop/api/CarbonTableInputFormat.java

```diff
@@ -389,14 +389,16 @@ protected FileSplit makeSplit(String segmentId, String filePath, long start, lon
 
   public void updateLoadMetaDataDetailsToSegments(List<Segment> validSegments,
       List<org.apache.carbondata.hadoop.CarbonInputSplit> prunedSplits) {
+    Map<String, Segment> validSegmentsMap = validSegments.stream()
```

Comment:
creating a map everytime for query is also overhead when the valid segments is in thousands, I suggest we can use `SET` for `validSegments` when it is formed originally instead of 'LIST'

Reply:
Appreciate for that good suggestion. But here we can only use Map instead of Set. 

Reason: The The pseudo-code for this function is shown as below.

        // **0. segments.hashcode is segmentno, to when we compare 2 segments, only segmentno will be compared**
        if (validSegments.contains(segmentInSplit)) {
		  **1. fetch the segment from validSegments** 
		  Segment segmentInValidSegment <- fetch the segment from validSegments
		  2. 
          segmentInSplit.setLoadMetadataDetails(segmentInValidSegment.getLoadMetadataDetails);
        }

For SET. it's hard to fetch the segment from validSegments.

Reply:
if you see the `equals` implementation of `Segment.java`, it is based on only segment number comparison. so, I think you can still use `SET`

Reply:
Yeah. But if use SET. it's hard to read a element with specified segmentno. 
In this code. we need to read a segment with specified segmentno from validSegments.

Reply:
oh, I got what you mean, we still need to get the element from valid segment to read its LoadMetadataDetails. So, SET can help only for contains, but not for getting the valid segment.